ext/Shared/__init__.lua

```diff
@@ -263,124 +51,129 @@ local function GetCustomLevel(p_LevelName, p_GameModeName)
 
-	if p_Partition == nil then
-		return
-	end
+	for l_PartitionGuid, l_Instances in pairs(m_CustomLevelData) do
+		for _, l_InstanceGuid in ipairs(l_Instances) do
+			-- print(tostring(l_PartitionGuid) .. ' - ' .. tostring(l_InstanceGuid))
 
-	local s_PrimaryInstance = p_Partition.primaryInstance
+			ResourceManager:RegisterInstanceLoadHandlerOnce(Guid(l_PartitionGuid), Guid(l_InstanceGuid), function(p_Instance) 
+				p_Instance = _G[p_Instance.typeInfo.name](p_Instance)
+				p_Instance:MakeWritable()
+				p_Instance.excluded = true
+			end)
+		end
+	end
+end)
 
-	if s_PrimaryInstance == nil then
-		print('Instance is null? ' .. p_Partition.name, true)
+---Patches the level, adding a SubWorldReferenceObjectData to the level that references the SubWorld in the custom bundle
+local function _PatchLevel(p_LevelName)
+	local s_Data = LevelData(ResourceManager:SearchForDataContainer(SharedUtils:GetLevelName()))
+	s_Data:MakeWritable()
+
+	local s_SWROD = SubWorldReferenceObjectData(Guid('6a724d44-4efd-4f7e-9249-2230121d7ecc'))
+	
+	s_SWROD.bundleName = BUNDLE_PREFIX .. '/' .. p_LevelName:gsub(".*/", "") .. '/' .. SharedUtils:GetCurrentGameMode()
+	s_SWROD.blueprintTransform = LinearTransform()
+	s_SWROD.blueprint = nil
+	s_SWROD.objectVariation = nil
+	s_SWROD.streamRealm = StreamRealm.StreamRealm_Both
+	s_SWROD.castSunShadowEnable = true
+	s_SWROD.excluded = false
+	s_SWROD.inclusionSettings = nil
+	s_SWROD.autoLoad = true
+	s_SWROD.isWin32SubLevel = true
+	s_SWROD.isXenonSubLevel = true
+	s_SWROD.isPs3SubLevel = true
+	s_SWROD.isEventConnectionTarget  = 2
+	s_SWROD.isPropertyConnectionTarget  = 3
+
+	local s_Partition = s_Data.partition
+
+	if not s_Partition then
+		print('Partition was nil', true)
 		return
 	end
 
-	-- if l_Instance:Is("Blueprint") then
-	--print("-------"..Blueprint(l_Instance).name)
-	-- end
+	s_Partition:AddInstance(s_SWROD)
 
-	if s_PrimaryInstance.typeInfo.name == "LevelData" then
-		local s_Instance = LevelData(s_PrimaryInstance)
+	local s_HighestIndexInPartition = 0
 
-		if s_Instance.name == SharedUtils:GetLevelName() then
-			print("----Registering PrimaryLevel guids")
-			s_Instance:MakeWritable()
+	for _, l_Instance in pairs(s_Partition.instances) do
+		if l_Instance:Is("GameObjectData") then
+			l_Instance = GameObjectData(l_Instance)
 
-			m_PrimaryLevelGuids = {
-				instanceGuid = s_Instance.instanceGuid,
-				partitionGuid = s_Instance.partitionGuid
-			}
-		end
-	elseif s_PrimaryInstance:Is('ObjectVariation') then
-		-- Store all variations in a map.
-		local s_Variation = ObjectVariation(s_PrimaryInstance)
-		m_ObjectVariations[s_Variation.nameHash] = s_Variation
-
-		if m_PendingVariations[s_Variation.nameHash] ~= nil then
-			for _, l_Object in pairs(m_PendingVariations[s_Variation.nameHash]) do
-				l_Object.objectVariation = s_Variation
+			if l_Instance.indexInBlueprint > s_HighestIndexInPartition and l_Instance.indexInBlueprint ~= 65535 then
+				s_HighestIndexInPartition = l_Instance.indexInBlueprint
```

Comment:
Maybe put this whole index thing in a separate function _GetHighestIndex. Your decision.